core/src/miner/miner.rs

```diff
@@ -366,6 +370,30 @@ impl Miner {
         }
         cdebug!(MINER, "Pushed {}/{} transactions", tx_count, tx_total);
 
+        let actions = self.engine.close_block_actions(open_block.block()).map_err(|e| {
+            warn!("Encountered error on closing the block: {}", e);
+            e
+        })?;
+        if !actions.is_empty() {
+            ctrace!(MINER, "Enqueue {} transactions to close block", actions.len());
+            // TODO: This generates a new random account to make the transaction.
+            // It should use the block signer.
+            let tx_signer = block_tx_signer.unwrap_or_else(Private::random);
+            let mut seq = block_tx_seq
+                .map(Ok)
+                .unwrap_or_else(|| open_block.state().seq(&public_to_address(&tx_signer.public_key())))?;
+            for action in actions {
+                let tx = Transaction {
+                    network_id: chain.network_id(),
+                    action,
+                    seq,
+                    fee: 0,
+                };
+                seq += 1;
+                let tx = VerifiedTransaction::new_with_sign(tx, &tx_signer);
+                open_block.push_transaction(tx, chain, parent_header.number(), parent_header.timestamp())?;
```

Comment:
Yeah, that's one of the problems that I couldn't solve yet.
Because we can know the size of close transactions after the execution, at this position, we cannot know the exact size.
Could I leave a comment and fix it later?

Reply:
It's a difficult problem.
I'll be happy if you leave a comment.

Reply:
I added the comment.